### ISSUE TITLE
Add support for ES dynamic `import()`

### ISSUE THREAD

**Issue by sjrd** [scala-js/scala-js#3686]
The ECMAScript proposal for dynamic imports at https://github.com/tc39/proposal-dynamic-import is [about to reach stage 4](https://github.com/tc39/ecma262/pull/1482) (supposedly it has reached stage 4 during the last TC meeting, but the PR hasn't been merged as of this writing). That means it's time for us to support it.

I would simply suggest a new primitive
```scala
def `import`(specifier: String): js.Primitive[scala.Any]
```
in the package object `js`, so that calls would look like
```scala
js.`import`("foobar.js")
```
It is a bit awkward that Scala imposes backticks to call that method, but I believe it's better than giving it a different name. In the future we might lobby for `import` not to be considered a keyword in Scala if it is followed by an open parenthesis.

---

Note that there is also a Stage 3 proposal for so-called `import.meta`, see https://github.com/tc39/proposal-import-meta. We might want to consider instead a
```scala
object `import` {
  def apply(specifier: String): js.Promise[scala.Any]
}
```
where `apply` is the primitive, so that we can later add a `def meta` in that object. If it's a `def import`, we might have trouble adding a "field" `meta` to it. Although I think Scala would not prevent us from having
```scala
def `import`(specifier: String): js.Promise[scala.Any]
// *and*
def `import`: ImportMeta
sealed trait ImportMeta {
  def meta: ...
}
```
but I never liked the overloading based on whether or not there is an *additional* parameter list :s

---

Another question is the result type of `js.import`: should it be:
* `js.Promise[scala.Any]`?
* `js.Promise[js.Any]`?
* `js.Promise[js.Dynamic]`?
* `js.Promise[js.ModuleInstance]` where `ModuleInstance` is *kind of* `js.Dynamic` but with less stuff (only read-only fields and methods)
* Or even parameterized with `[A <: js.Any]` and returning `js.Promise[A]`?

`scala.Any` is the most conservative one. `A` is the most liberal, allowing to avoid `asInstanceOf`s (but hiding the cast inside the method, in a sense, although it always succeed because `A <: js.Any` anyway).

**Comment by gzm0** [comment]
Two things:

Why do we need to be concerned about `import.meta`? If this is a primitive, isn't it enough if everything is source compatible? So we can still switch to `object import` with `def apply` later if needed.

I think we should specify this as:

```scala
 def `import`[A <: js.Any](specifier: String): js.Promise[A]
```

Since this will: Make the typed use case easy and force the dynamically typed use case to be explicit:

```scala
js.`import`[Foo]("foo.js")
js.`import`[js.Dynamic]("foo.js")
```

**Comment by sjrd** [comment]
Ah, good point about `import.meta`. Indeed we can change it later.

> Since this will: Make the typed use case easy and force the dynamically typed use case to be explicit:

I like that rationale :)

**Comment by sjrd** [comment]
Hum actually we do need to care about `import.meta`, because we cannot prevent a compiler version `x` from being used with a scalajs-library version `y > x`. So if we move `js.import` from a `def import` in `js` to a `def apply` in `object import`, the compiler won't consider the `import.apply` as a primitive, and things will break.

We don't really have this problem when we introduce *new* primitives, usually, because when we do that, we also bump the minor IR version number, which forces the sbt plugin to be upgraded, which also forces the compiler to be upgraded.

In this case we wouldn't need to bump the IR version number, so we wouldn't have this protection.

Given that `import.meta` is going to happen at some point (it's Stage 3, it's very unlikely to be completely cancelled), I think we should already design for its later introduction.

**Comment by sjrd** [comment]
Fixed in 17407e74cbe7ad06981bc5e77e019252681f19b2.
